core/store/src/db.rs

```diff
@@ -76,6 +77,7 @@ impl DBTransaction {
     }
 
     pub(crate) fn delete(&mut self, col: DBCol, key: Vec<u8>) {
+        log_assert!(!col.is_rc(), "{col}");
```

Comment:
I’d add it to StoreUpdate::delete rather than here to be more consistent. All other checks are in StoreUpdate and none ar here.

Reply:
Ah, I actually intended to delete this, to make sure I don't introduce new asserts. Missed this one though, thanks!